libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -132,48 +136,71 @@
      * @param authTokenProvider
 	 */
 	public RestClient(ClientInfo clientInfo, String authToken, HttpAccess httpAccessor, AuthTokenProvider authTokenProvider) {
-		this(httpAccessor, new OAuthRefreshInterceptor(clientInfo, authToken, authTokenProvider));
-	}
-
-	public RestClient(HttpAccess httpAccessor, OAuthRefreshInterceptor httpInterceptor) {
+		this.clientInfo = clientInfo;
```

Comment:
The freshly created oauth refresh interceptor ends up not being used if we find a cached ok http client.